### PR TITLE
Fix python-ldap not throwing exception on anonymous binds when it should

### PR DESCRIPTION
**Pull request by whisperity**
See for further details: https://mail.python.org/pipermail/python-ldap/2012q4/003183.html

### CODE REVIEW

File: libcodechecker/libauth/cc_ldap.py [hunk 1]
```diff
@@ -218,14 +218,27 @@ def __init__(self, ldap_config, who=None, cred=None):
...
                     res = self.connection.simple_bind_s()
                     LOG.debug(res)
                 else:
-                    # Bind with the given credentials.
-                    LOG.debug(who)
+                    LOG.debug("Binding with credential: " + who)
                     res = self.connection.simple_bind_s(who, cred)
+                    whoami = self.connection.whoami_s()
                     LOG.debug(res)
+                    LOG.debug(whoami)
```
Comment (gyorb) [inline libcodechecker/libauth/cc_ldap.py]: Could this leak some sensitive information to the logs?

Comment (whisperity) [reply libcodechecker/libauth/cc_ldap.py]: No. It is either `None` or the username, which is already logged above.

**Comment by whisperity** [comment]
@gyorb Added, and also explained it a bit in more detail. Having mailing list links in the code is not too good, what if the mailing list goes out of business?

**Comment by gyorb** [comment]
@whisperity yes I know it is possible that the mailing list will not be accessible but I still prefer the idea that most of the information should be in the source.
I'm not sure if there will be any other problem with ldap authentication everybody will search for the pull request in github to read the comments here.

**Pull request merged**